src/main/java/com/atlassian/jira/cloud/jenkins/util/XmlUtils.java

```diff
@@ -0,0 +1,49 @@
+package com.atlassian.jira.cloud.jenkins.util;
+
+import hudson.XmlFile;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.StringReader;
+
+/**
+ * Utility class for handling XML-related operations such as parsing XML files and extracting values
+ * from XML elements.
+ *
+ * <p>This class provides methods to parse XML files, extract specific values based on tag names,
+ * and handle boolean values stored within XML elements.
+ */
+public class XmlUtils {
+
+    public Element parseXmlFile(final XmlFile file)
+            throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
Just did some reading on this and DocumentBuilder. I think with the current set up we're initialising and creating a new object for every for parsing operation. Also, looks like DocumentBuilder isn't thread safe so could potentially run into race conditions or other gnarly thread safety issues. 

Maybe we could refactor this so we reuse the DocumentBuilder instance instead of creating a new one for each XML file. Apparently creating a new one is a pretty expensive option so having just the one instance would be more efficient. Something like this (haven't tested obviously so could completely break everything :badpokerface:

```
private final DocumentBuilder builder;

public XmlUtils() throws ParserConfigurationException {
    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
    this.builder = factory.newDocumentBuilder();
}

public Element parseXmlFile(final XmlFile file) throws IOException, SAXException {
    Document document = builder.parse(new InputSource(new StringReader(file.asString())));
    return document.getDocumentElement();
}
```

Reply:
i agree with this strongly